src/vs/workbench/services/configuration/browser/configurationService.ts

```diff
@@ -1351,6 +1351,7 @@ class ConfigurationDefaultOverridesContribution extends Disposable implements IW
 		super();
 
 		this.updateDefaults().then(() => {
+			this._register(workbenchAssignmentService.onDidRefetchAssignments(() => this.processExperimentalSettings(this.autoExperimentalSettings, true)));
```

Comment:
Just to note, this could leak in tests, if `updateDefaults` resolves after the service already got disposed!